cmd/boost/direct_deal.go

```diff
@@ -0,0 +1,440 @@
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/big"
+	"github.com/filecoin-project/go-state-types/builtin"
+	datacap2 "github.com/filecoin-project/go-state-types/builtin/v9/datacap"
+	verifregst "github.com/filecoin-project/go-state-types/builtin/v9/verifreg"
+	lapi "github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/actors"
+	"github.com/filecoin-project/lotus/chain/actors/builtin/datacap"
+	"github.com/filecoin-project/lotus/chain/types"
+	lcli "github.com/filecoin-project/lotus/cli"
+	"github.com/ipfs/go-cid"
+	"github.com/urfave/cli/v2"
+)
+
+var directDealAllocate = &cli.Command{
+	Name:  "allocate",
+	Usage: "Create new allocation[s] for verified deals",
+	Flags: []cli.Flag{
+		&cli.StringSliceFlag{
+			Name:     "miner",
+			Usage:    "storage provider address[es]",
+			Required: true,
+			Aliases:  []string{"m", "provider", "p"},
+		},
+		&cli.StringSliceFlag{
+			Name:     "pieceInfo",
+			Usage:    "data pieceInfo[s] to create the allocation. The format must be pieceCid1=pieceSize1 pieceCid2=pieceSize2",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:  "wallet",
+			Usage: "the wallet address that will used create the allocation",
+		},
+		&cli.BoolFlag{
+			Name:  "quiet",
+			Usage: "do not print the allocation list",
+			Value: false,
+		},
+	},
+	Before: before,
+	Action: func(cctx *cli.Context) error {
+		ctx := bcli.ReqContext(cctx)
+
+		n, err := clinode.Setup(cctx.String(cmd.FlagRepo.Name))
+		if err != nil {
+			return err
+		}
+
+		api, closer, err := lcli.GetGatewayAPI(cctx)
+		if err != nil {
+			return fmt.Errorf("cant setup gateway connection: %w", err)
+		}
+		defer closer()
+
+		// Get wallet address from input
+		walletAddr, err := n.GetProvidedOrDefaultWallet(ctx, cctx.String("wallet"))
+		if err != nil {
+			return err
+		}
+
+		log.Debugw("selected wallet", "wallet", walletAddr)
+
+		// Get all minerIDs from input
+		maddrs := make(map[abi.ActorID]lapi.MinerInfo)
+		minerIds := cctx.StringSlice("miner")
+		for _, id := range minerIds {
+			maddr, err := address.NewFromString(id)
+			if err != nil {
+				return err
+			}
+
+			// Verify that minerID exists
+			m, err := api.StateMinerInfo(ctx, maddr, types.EmptyTSK)
+			if err != nil {
+				return err
+			}
+
+			mid, err := address.IDFromAddress(maddr)
+			if err != nil {
+				return err
+			}
+
+			maddrs[abi.ActorID(mid)] = m
+		}
+
+		// Get all pieceCIDs from input
+		rDataCap := big.NewInt(0)
+		var pieceInfos []*abi.PieceInfo
+		pieces := cctx.StringSlice("pieceInfo")
+		for _, p := range pieces {
+			pieceDetail := strings.Split(p, "=")
+			if len(pieceDetail) > 2 {
+				return fmt.Errorf("incorrect pieceInfo format: %s", pieceDetail)
+			}
+
+			n, err := strconv.ParseInt(pieceDetail[1], 10, 64)
+			if err != nil {
+				return fmt.Errorf("failed to parse the piece size for %s: %w", pieceDetail[0], err)
```

Comment:
```suggestion
				return fmt.Errorf("failed to parse the piece size for %s: %w", pieceDetail[1], err)
```

Reply:
I have clarified the error a little further. I was printing pieceCid on purpose. I have modified it to print both cid and size now.